src/legacy/core_plugins/metrics/public/kbn_vis_types/request_handler.js

```diff
@@ -20,47 +20,90 @@
 import { validateInterval } from '../lib/validate_interval';
 import { timezoneProvider } from 'ui/vis/lib/timezone';
 import { timefilter } from 'ui/timefilter';
+import { kfetchAbortable } from 'ui/kfetch';
+
+const ABORT_REQUEST_CODE = 20;
+const API_PATH_NAME = '../api/metrics/vis/data';
 
 const MetricsRequestHandlerProvider = function (Private, Notifier, config, $http, i18n) {
   const notify = new Notifier({ location: i18n('tsvb.requestHandler.notifier.locationNameTitle', { defaultMessage: 'Metrics' }) });
+  const requestsMap = new Map();
+
+  const handleError = (error, callback) => {
+    if (error) {
+      notify.error(error);
+    }
+    callback({});
+  };
```

Comment:
Shouldn't we also need to remove the mapped `abort` function for the faulty request?

Reply:
we should, my issue. Fixed